generative/maisi/scripts/augmentation.py

```diff
@@ -0,0 +1,555 @@
+        while True:
+            tumor_mask = tumor_mask_
+            # apply random augmentation
+            distorted_mask = elastic((tumor_mask == 2), spatial_size=spatial_size).as_tensor()
+
+            # generate final tumor mask
+            count += 1
+            tumor_mask = finalize_tumor_mask(distorted_mask, organ_mask, tumor_size * 0.80)
+            if tumor_mask is not None:
+                break
+            if count > MAX_COUNT:
+                raise ValueError("Please check if pancreas tumor is inside pancreas.")
+    else:
+        tumor_mask = tumor_mask_
+
+    volume[tumor_mask == 1] = tumor_label[0]
+
+    return volume.unsqueeze(0).to(device)
+
+
+def augmentation_colon_tumor(whole_mask: Tensor, spatial_size: tuple[int, int, int] | int | None = None) -> Tensor:
+    """
+    Colon tumor augmentation.
+
+    Args:
+        whole_mask: input 3D multi-label mask, [1,1,H,W,D] torch tensor.
+        spatial_size: output image spatial size, used in random transform.
+                      If not defined, will use (H,W,D). If some components are non-positive values,
+                      the transform will use the corresponding components of whole_mask size.
+                      For example, spatial_size=(128, 128, -1) will be adapted to (128, 128, 64)
+                      if the third spatial dimension size of whole_mask is 64.
+
+    Return:
+        augmented mask, with shape of spatial_size and data type as whole_mask.
+
+    Example:
+
+        .. code-block:: python
+
+            # define a multi-label mask
+            whole_mask = torch.zeros([1,1,128,128,128])
+            whole_mask[0,0, 90:110, 90:110, 90:110]=62
+            whole_mask[0,0, 97:103, 97:103, 97:103]=27
+            augmented_whole_mask = augmentation_colon_tumor(whole_mask)
+    """
+    # Initialize binary tumor mask
+    device = whole_mask.device
+    volume = whole_mask.squeeze(0).cuda() if not whole_mask.is_cuda else whole_mask.squeeze(0)
+    tumor_label = [27]
+    tumor_mask_ = initialize_tumor_mask(volume, tumor_label)
+
+    # Define augmentation transform
+    elastic = Rand3DElastic(
+        mode="nearest",
+        prob=1.0,
+        sigma_range=(5, 8),
+        magnitude_range=(100, 200),
+        translate_range=(5, 5, 5),
+        rotate_range=(np.pi / 36, np.pi / 36, np.pi / 36),
+        scale_range=(0.1, 0.1, 0.1),
+        padding_mode="zeros",
+    )
+
+    tumor_size = torch.sum(tumor_mask_)
+    ###########################
+    # before move tumor maks, full the original location by organ labels
+    volume[tumor_mask_.bool()] = 62
+    ###########################
+    if tumor_size > 0:
+        # get organ mask
+        organ_mask = (volume == 62).float()
+        organ_mask = dilate_one_img(organ_mask.squeeze(0), filter_size=5, pad_value=1.0)
+        organ_mask = erode_one_img(organ_mask, filter_size=5, pad_value=1.0).unsqueeze(0)
```

Comment:
I would suggest directly using dilate and erode function here.

Reply:
Thank you for the suggestion. But so many code are using it...Also Vista will use it...I'd rather not change...

Reply:
I didn't see any risk here to use the erode and dilate function you have already added in the core. `dilate_one_img` and `erode_one_img` only add unsqueeze and squeeze.
But I will merge this PR first, we can leave it later.